lib/client.js

```diff
@@ -50,20 +50,7 @@ function connect (client) {
   })
 }
 
-function destroyMaybe (client) {
-  if (
-    client.closed &&
-    !client._lastBody &&
-    client[kQueue].length() === 0 &&
-    client[kCallbacks].length === 0
-  ) {
-    client.destroy()
-  }
-}
-
 function reconnect (client, err) {
-  err = err || new Error('premature close')
```

Comment:
err can never be falsy, https://github.com/mcollina/undici/pull/34/files#diff-50cfa59973c04321b5da0c6da0fdf4feL49